jni/README.md

```diff
@@ -31,5 +23,3 @@ Using this library assumes some familiarity with JNI - it's threading model and
 
 This library is a part of `jnigen` - a 2022 GSoC project.
```

Comment:
I think we can get rid of this.

The GSoC relatedness can be found on GSoC and our blogposts. (And the commit history 🎉 )

For users of this package it's not important to know how it started.